### PR TITLE
Change Request.Client to ClientConn

### PR DESCRIPTION
**Pull request by cizmazia**
- Changed Request.Client from ClientCommander to ClientConn

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -215,6 +215,10 @@ func (c *Client) DialWithContext(ctx context.Context, address string) (clientCon
 	return clientConn, nil
 }
 
+func (co *ClientConn) NetworkSession() networkSession {
```
Comment (jkralik) [inline client.go]: We dont want to expose network session from library

**Comment by cizmazia** [comment]
merged

File: client.go [hunk 2]
```diff
@@ -350,8 +354,14 @@ func (co *ClientConn) ObserveWithContext(ctx context.Context, path string, obser
 
 // Close close connection
 func (co *ClientConn) Close() error {
-	co.srv.Shutdown()
-	<-co.shutdownSync
+	if co.srv != nil {
+		co.srv.Shutdown()
+	} else {
+		co.commander.Close()
```
Comment (jkralik) [inline client.go]: pls check return error and forward to caller

File: client.go [hunk 3]
```diff
@@ -354,15 +354,16 @@ func (co *ClientConn) ObserveWithContext(ctx context.Context, path string, obser
...
+	var err error
 	if co.srv != nil {
-		co.srv.Shutdown()
+		err = co.srv.Shutdown()
 	} else {
-		co.commander.Close()
+		err = co.commander.Close()
 	}
 	if co.shutdownSync != nil {
 		<-co.shutdownSync
```
Comment (jkralik) [inline client.go]: and this error?

**Comment by cizmazia** [comment]
rebased

**Comment by cizmazia** [comment]
merged

**Pull request closed without merging**